### PR TITLE
Remove support for OTEL spanmetrics processor

### PR DESCRIPTION
**Pull request by varshith257**
## Which problem is this PR solving?
- Part of #4736 

## Description of the changes
-   removed support for the spanmetrics processor flag.

## How was this change tested?
-  CI

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/collector/app/metrics.go [hunk 1]
```diff
@@ -59,28 +59,28 @@ func initOtherServicesSamplers() map[model.SamplerType]string {
 }
 
 // SpanProcessorMetrics contains all the necessary metrics for the SpanProcessor
-type SpanProcessorMetrics struct {
```
Comment (yurishkuro) [inline cmd/collector/app/metrics.go:62]: this class has nothing to do with the ticket. SpanMetricsProcessor refers to an OTEL Collector component.

Comment (varshith257) [reply cmd/collector/app/metrics.go:62]: Yes, got it. At the time of this PR raised I thought it is as a Jaeger component. After working ony proposal for SPM, I got clear context of it.

Now, I will remove that flag and corresponding code and will make default of spanmetrics connector and add it the same in Readme

**Comment by varshith257** [comment]
@yurishkuro I am thinking of a better migration guide that will be eradicated in v2. Till then it may help others who are still referencing processor

**Comment by varshith257** [comment]
@yurishkuro PTAL and if any changes needed I will incorporate

**Comment by yurishkuro** [comment]
no actual code is remove here

**Comment by varshith257** [comment]
Does it mean spanmetrics processor code or it's flag code?

**Comment by varshith257** [comment]
@yurishkuro I have found Spanmetrics Processor instances in metrics.go. Does it needed to be migrated to Spanmetrics Connector or simply remove it?

I think if we remove it how new instance of the SpanProcessor Metrics struct will be initialised?

**Comment by varshith257** [comment]
The lint check is failing due to other commits added while rebasing earlier before the new DCO method is applied. I think it has been standardized. Will close this PR and submit a new one instead

**Pull request closed without merging**